reviews/module.py

```diff
@@ -0,0 +1,314 @@
+                for chunk in review_chunks[1:]:
+                    embed.add_field(name="", value=chunk)
+            embeds.append(embed)
+        await ScrollableVotingEmbed(ctx, embeds).scroll()
+
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.guild_only()
+    @review.command(name="list", aliases=["available"])
+    async def review_list(self, ctx: discord.ext.commands.Context):
+        """Get list of reviewed subjects"""
+        subjects = Subject.get_reviewed(ctx.guild)
+        if len(subjects) == 0:
+            await ctx.reply(_(ctx, "There are no rated subjects yet."))
+            return
+        embed = discord.Embed(title=_(ctx, "Available subjects"))
+        embed.add_field(
+            name="", value=", ".join(subject.shortcut for subject in subjects)
+        )
+        await ctx.reply(embed=embed)
+
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.guild_only()
+    @review.command(name="my-list")
+    async def review_my_list(self, ctx: discord.ext.commands.Context):
+        """Get list of your reviewed subjects."""
+        subjects = Subject.get_reviewed_by_user(ctx.guild, ctx.author)
+        if len(subjects) == 0:
+            return await ctx.reply(_(ctx, "You have not rated any subject yet."))
+        embed = discord.Embed(title=_(ctx, "My rated subjects"))
+        embed.add_field(
+            name="", value=", ".join(subject.shortcut for subject in subjects)
+        )
+        await ctx.reply(embed=embed)
+
+    @staticmethod
+    async def _add_review(
+        ctx: discord.ext.commands.Context,
+        subject: str,
+        mark: int,
+        text: str,
+        anonymous: bool,
+    ):
+        """Add and return review"""
+        if mark < 1 or mark > 5:
+            await ctx.reply(_(ctx, "Mark must be in the range <1, 5>."))
+            return None
+
+        # check if subject is in database
+        db_subject = Subject.get(ctx.guild, subject)
+        if db_subject is None:
+            await ctx.reply(_(ctx, "Unknown subject."))
+            return None
+
+        if text is None or not len(text):
+            await ctx.reply(_(ctx, "Please provide text for the review."))
+            return None
+
+        result = Review.add(ctx.guild, ctx.author, subject, mark, anonymous, text)
+        return result
+
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.guild_only()
+    @review.command(name="add", aliases=["update"])
+    async def review_add(
+        self, ctx: discord.ext.commands.Context, subject: str, mark: int, *, text: str
+    ):
+        """Add a review
+
+        subject: Subject code
+        mark: 1-5 (one being best)
+        text: Your review
+        """
+        result = await self._add_review(ctx, subject, mark, text, False)
+        if result is not None:
+            await guild_log.info(
+                ctx.author, ctx.channel, f"Added/updated review for subject {subject}."
+            )
+            await ctx.reply(_(ctx, "Review successfully added."))
```

Comment:
When calling `!review add` without  parameters, help is **not** shown. It should be right? Now it shows only User input error

Reply:
This also applies to command `!review add-anonymous` where it needs to be  imo fixed. This error is showing even with commands like `!review remove`, but there is not really a reason to fix it there since these commands have only one parameter.